modules/puppet_proxy/class_scanner_eparser.rb

```diff
@@ -44,7 +44,11 @@ def initialize
       end
 
       def do_find ast
-        @@finder_visitor.visit_this(self, ast)
+        if Puppet::PUPPETVERSION.to_i < 4
+          @@finder_visitor.visit_this(self, ast)
+        else
+          @@finder_visitor.visit_this(self, ast, [])
+        end
```

Comment:
This conditional can be removed alltogether if you: change the constructor to something like:

```
      def initialize
        @klasses = []
        @finder_visitor = Puppet::Pops::Visitor.new(self,'find',0,0)
      end
```

and then update `do_find` method to:
```
      def do_find ast
        @finder_visitor.visit(ast)
      end
```

Note that I replaced class-variable `@@finder_visitor` with an instance variable `@finder_visitor`.